code/computational_geometry/src/chans_algorithm/chans_algorithm.cpp

```diff
@@ -0,0 +1,249 @@
+    #include <iostream>
+    #include <stdlib.h>
+    #include <vector>
+    #include <algorithm>	// For qsort() algorithm
+    #include <utility>		// For pair() STL
+    #define RIGHT_TURN -1  // CW
+    #define LEFT_TURN 1  // CCW
+    #define COLLINEAR 0  // Collinear
+    using namespace std;
+    /*
+    Class to handle the 2D points!
+    */
+    class point
+    {
+    public:
+        int x;
+        int y;
+        point (int newx=0,int newy=0){
+            x=newx;
+            y=newy;
+        }
+        /*
+            Overloaded == operator to check for equality between 2 objects of class point
+        */
+        friend bool operator== (const point& p1,const point& p2){
+            return (p1.x==p2.x && p1.y==p2.y);
```

Comment:
No need for parentheses.

Reply:
Tried removing curly braces but there it's giving error (precedence of operators).